blockchain/blockchain.go

```diff
@@ -1099,96 +1100,97 @@ type BlockSignFunc func(blockHash, stateDiffCommitment *felt.Felt) ([]*felt.Felt
 
 // Finalise will calculate the state commitment and block hash for the given pending block and append it to the
 // blockchain
-func (b *Blockchain) Finalise(pending *Pending, sign BlockSignFunc) error {
+func (b *Blockchain) Finalise(txs []core.Transaction, receipts []*core.TransactionReceipt, diff *core.StateDiff,
```

Comment:
Wdty about either:
1. renaming `Finalise() `to (eg) `BuildAndStorePending()`?
2. removing `Finalise() , creating a `BuildPending()` funciton, and just calling `b.finalise()` on that?

Reply:
I think a renaming refactor could be done outside this PR. As long as the name reflects that the pending block is appended to the chain, I'm good.